### DISCUSSION TITLE
Is there a way to load las file(point cloud data) as Tile3DLayer on the map?

### DISCUSSION THREAD

**Discussion by daqjjang** [visgl/deck.gl#5279] [Q&A]
Hi, I am just a beginner with Web and Deck.gl .
I just wonder whether there is a way to load las file on the map.

import React, {useState} from 'react';
import {render} from 'react-dom';
import {StaticMap} from 'react-map-gl';
import DeckGL from '@deck.gl/react';
import {Tile3DLayer} from '@deck.gl/geo-layers';
import {COORDINATE_SYSTEM} from '@deck.gl/core';
import {CesiumIonLoader} from '@loaders.gl/3d-tiles';
import {Tiles3DLoader} from '@loaders.gl/3d-tiles';
import {PointCloudLayer} from '@deck.gl/layers';
import {LASWorkerLoader} from '@loaders.gl/las';
// import {PLYWorkerLoader} from '@loaders.gl/ply';
import {registerLoaders} from '@loaders.gl/core';

// Additional format support can be added here, see
// https://github.com/visgl/loaders.gl/blob/master/docs/api-reference/core/register-loaders.md
registerLoaders(LASWorkerLoader);
// registerLoaders(PLYWorkerLoader);

// Data source: kaarta.com
const LAZ_SAMPLE =
'http://localhost:8080/metuArch.las';
const layers3d = [
new PointCloudLayer({
id: 'laz-point-cloud-layer',
data: LAZ_SAMPLE,
coordinateOrigin: [ 126.97, 37.56],
coordinateSystem: COORDINATE_SYSTEM.METER_OFFSETS,
getNormal: [0, 1, 0],
getColor: [255, 255, 255],
opacity: 0.5,
pointSize: 0.5
})
];
return (

);
}

export function renderToDOM(container) {
render(, container);
}

**Comment by ibgreen** [comment]
If you are loading a single LAS file you want to use the `PointCloudLayer`. Check out e.g. https://loaders.gl/examples/pointcloud and https://github.com/visgl/loaders.gl/tree/master/examples/website/pointcloud.

Also, if you are displaying the LAS file on the map you need to position and scale your point cloud correctly. Perhaps using `Layer.coordinateOrigin` prop. loaders/deck does not read coordinate system information from LAS files.

**Reply by daqjjang** [reply]
Thanks for the help!
Actually, las file size is big, so that is why I try to use Tile3DLayer without uploading the file on Cesium.

**Reply by ibgreen** [reply]
That won't work. 3DTileLayer expects files to be divided into tiles before you start loading them.

**Reply by daqjjang** [reply]
Oh, I see.
Then is there way to convert las file to 3d tiles format?

**Reply by ibgreen** [reply]
If you don't want to use Cesium, you need to look around for open source tiling solutions. Perhaps the tool referenced in https://github.com/visgl/deck.gl/issues/4898.

**Reply by daqjjang** [reply]
Thanks for the help!
